src/internal/operators/bufferTime.ts

```diff
@@ -69,189 +79,128 @@ export function bufferTime<T>(bufferTimeSpan: number, bufferCreationInterval: nu
  * @return {Observable<T[]>} An observable of arrays of buffered values.
  * @name bufferTime
  */
-export function bufferTime<T>(bufferTimeSpan: number): OperatorFunction<T, T[]> {
-  let length: number = arguments.length;
-
+export function bufferTime<T>(bufferTimeSpan: number, ...otherArgs: any[]): OperatorFunction<T, T[]> {
   let scheduler: SchedulerLike = async;
-  if (isScheduler(arguments[arguments.length - 1])) {
-    scheduler = arguments[arguments.length - 1];
-    length--;
-  }
 
-  let bufferCreationInterval: number | null = null;
-  if (length >= 2) {
-    bufferCreationInterval = arguments[1];
+  if (isScheduler(otherArgs[otherArgs.length - 1])) {
+    scheduler = otherArgs.pop() as SchedulerLike;
   }
 
-  let maxBufferSize: number = Infinity;
-  if (length >= 3) {
-    maxBufferSize = arguments[2];
-  }
+  const bufferCreationInterval = (otherArgs[0] as number) || null;
+  const maxBufferSize = (otherArgs[1] as number) || Infinity;
```

Comment:
This is breaking since before if `length >= 3` it would be set to `0` but here it's set to `Infinity` in that case. `??` may help here

Reply:
I don't quite follow. [It appears that we were setting it to `Infinity` by default](https://github.com/ReactiveX/rxjs/blob/6416935ab03b38d7048d0641c8826ba5311eae8c/src/internal/operators/bufferTime.ts#L86-L89), and pulling the number out of the arguments if it was there. This is equivalent, AFAICT.